test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -87,10 +88,15 @@ private void assertCommandBehavior(String inputCommand,
             assertEquals(lastShownList, r.getRelevantPersons().get());
         }
 
+        //Parse Command
+        Command command = new Parser().parseCommand(inputCommand);
+
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if (command.isMutating()) {
```

Comment:
I think there should be a specific check to see if the command is a mutating command first. Then this check makes sense from a test perspective.